hal/tests/TESTS/mbed_hal_fpga_ci_test_shield/analogin/CMakeLists.txt

```
@@ -3,6 +3,16 @@
 
 include(mbed_greentea)
 
+if(NOT "DEVICE_ANALOGIN=1" IN_LIST MBED_TARGET_DEFINITIONS)
+    set(TEST_SKIPPED "Analog in not supported for this target")
+elseif(NOT "COMPONENT_FPGA_CI_TEST_SHIELD=1" IN_LIST MBED_TARGET_DEFINITIONS)
+    set(TEST_SKIPPED "FPGA CI Test Shield is needed to run this test")
+elseif(NOT("TARGET_FF_ARDUINO" IN_LIST MBED_TARGET_DEFINITIONS OR
+    "TARGET_FF_ARDUINO_UNO" IN_LIST MBED_TARGET_DEFINITIONS) AND
+    NOT DEFINED MBED_CONF_TARGET_DEFAULT_FORM_FACTOR)
```

```suggestion
    NOT "MBED_CONF_TARGET_DEFAULT_FORM_FACTOR" IN_LIST MBED_TARGET_DEFINITIONS)
```
It's an Mbed configuration macro, not a CMake variable. You can only use `DEFINED` on CMake variables. Same comments for the other files.